src/main/java/gov/nist/csd/pm/pap/GraphAdmin.java

```diff
@@ -56,14 +59,31 @@ public Node createPolicyClass(String name, Map<String, String> properties) throw
         // assign Super U to PC UA
         // getPAP().getGraphPAP().assign(superPolicy.getSuperU().getID(), pcUANode.getID());
         // assign superUA and superUA2 to PC
-        graph.assign(superPolicy.getSuperUserAttribute().getName(), pcNode.getName());
-        graph.assign(superPolicy.getSuperUserAttribute2().getName(), pcNode.getName());
+        if (superPolicy.getSuperUserAttribute() != null) {
+            if (graph.exists(superPolicy.getSuperUserAttribute().getName())) {
+                graph.assign(superPolicy.getSuperUserAttribute().getName(), pcNode.getName());
+
+            } else {
+                graph.createNode("super_ua1", UA, Node.toProperties(NAMESPACE_PROPERTY, "super"), pcNode.getName());
+            }
+        }
+
+        if (graph.exists(superPolicy.getSuperUserAttribute2().getName())) {
+            graph.assign(superPolicy.getSuperUserAttribute2().getName(), pcNode.getName());
+
+        } else {
+            graph.createNode("super_ua2", UA, Node.toProperties(NAMESPACE_PROPERTY, "super"), pcNode.getName());
+
+        }
```

Comment:
Do we need to create the super_ua1 and super_ua2 here? Aren't those created in the superPolicy.configure method?